### PR TITLE
fix(swap): round values for very small values in fiat and crypto

### LINKED ISSUE #21415: When entering a value with many decimals the value is incorrectly rounded during swap flow
### Steps:
1. Go to swap
2. Enter value with a lot of decimals (example: 0.000000000001 DAI)
3. Go to check rounding

### Actual result:
0 is shown for the following elements:
On swap page:
Image

On slippage screen:
Image

On swap approving and swap confirmation screens
Image

### Expected result:
For very small values that cannot be fully shown, the system should display the value in the form of:
- `<$0.01` for fiat or
- `<0.000000001 DAI` for token values.

### OS: 
IOS, Android
### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17

### LINKED ISSUE #21419: An random error occurs when long-tapping on an asset  from the wallet main page and selecting "Swap"
Unfortunately, this issue doesn't happen consistently, and there are no exact steps to reproduce it. Hopefully, the logs will help

### Steps:
1. Go to wallet main page
2. Long tap asset (in my case WBTC)
3. Tap "Swap"

### Actual result:
"cannot read property “toLowerCase” " of null error is shown

https://github.com/user-attachments/assets/2d42b405-5ba1-4bc7-aa7e-9349b40c4330

### Expected result:
No error is shown

### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17
### Env:
Nightly 11 Oct, 2024

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #21415
fixes #21419

### Summary

This PR fixes fiat values shown as 0 when the value is less than 0.01, and also the same for crypto tokens when the value is less than 0.000001. Also fixes an error when long-tapping on an asset from the wallet main page and selecting "Swap".

Screenshot 2024-10-15 at 6 28 40 PM
Screenshot 2024-10-15 at 7 38 31 PM
Screenshot 2024-10-15 at 6 29 05 PM

#### Platforms

- Android
- iOS

##### Functional

- wallet / transactions

#### Issue #21415
- Go to swap
- Enter value with a lot of decimals (example: 0.000000000001 DAI)
- Go to check rounding

#### Issue #21419
- Go to wallet main page
- Long tap asset (in my case WBTC)
- Tap "Swap"
- Verify no error is shown

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/common/utils.cljs [hunk 1]
```diff
@@ -165,6 +165,33 @@
...
+      (str currency-symbol fiat-value))))
+
+(defn sanitized-token-amount-to-display
+  "Formats a token amount to a specified number of decimals.
+  Returns a threshold value if the formatted amount is less than the minimum value represented by the decimals."
+  [amount display-decimals]
+  (let [number                (or (money/bignumber amount)
+                                  (money/bignumber 0))
+        amount-fixed-decimals (number/to-fixed number display-decimals)]
+    (if (and (= amount-fixed-decimals "0")
```
Comment (alwx) [inline src/status_im/contexts/wallet/common/utils.cljs:189]: why `"0"` and not `0`?

File: src/status_im/contexts/wallet/swap/setup_swap/view.cljs [hunk 2]
```diff
@@ -26,6 +26,7 @@
     [utils.string :as utils.string]))
 
 (def ^:private default-text-for-unfocused-input "0.00")
+(def ^:private default-token-symbol "ETH")
```
Comment (alwx) [inline src/status_im/contexts/wallet/swap/setup_swap/view.cljs:29]: I wonder if we don't have this constant somewhere already

**Review by alwx — APPROVED** [review]
Approved but added two small comments

File: src/status_im/contexts/wallet/common/utils.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im/contexts/wallet/common/utils.cljs:189]: That's because `number/to-fixed` calls `remove-trailing-zeroes` fn which returns a string

**Review by mohsen-ghafouri — APPROVED** [review]
LGTM

**Comment by Horupa-Olena** [comment]
@briansztamfater Thank you for your PR. 
Regarding the symbols, everything is fine.
As for the long tap bug, I tried different combinations under various conditions but couldn’t reproduce the issue. So, we’ll assume it's fixed.

You can go ahead and merge this PR.

**Pull request merged**